### ISSUE TITLE
Generate training loss nan

### ISSUE THREAD

**Issue by tobytyx** [wenhuchen/HDSA-Dialog#5]
Hello! When I try to train Response Generator by running
`CUDA_VISIBLE_DEVICES=0 python3.5 train_generator.py --option train --model BERT_dim128_w_domain_exp --batch_size 512 --max_seq_length 50 --field`.
The loss printed becomes nan.  What can I do to avoid this? Thank you!
<img width="819" alt="nan" src="https://user-images.githubusercontent.com/26656532/63841350-3f504700-c9b5-11e9-872a-eb5fb9d50e9c.PNG">

**Comment by wenhuchen** [comment]
Thank you so much for your feedback. I never encountered this problem before, here are some of my advices:

1. Can you reload the provided trained model to see if it can achieve the score reported (23.6 BLEU)?
2. Can you check if your pytorch version is exactly 1.0.0.
3. Can you set a smaller learning rate to see if it still happens.

We can discuss after you have tried the above methods.

**Comment by wenhuchen** [comment]
Hi, have you fixed your problem yet?

**Comment by tobytyx** [comment]
> Hi, have you fixed your problem yet?

Thank you for your advice! The score is 23.6 correctly.  I'll try your advices recently!

**Comment by tobytyx** [comment]
> Hi, have you fixed your problem yet?

I've got a question about your BLEU calculate function. When I use nltk.translate.bleu_score.corpus_bleu to test the result, it can only get 21.306 score., not 23.6 score?

**Comment by wenhuchen** [comment]
Yep, I know that problem. I inherited the evaluation script from https://github.com/budzianowski/multiwoz, their script appends a start token and an end token to the generated sentence "https://github.com/budzianowski/multiwoz/blob/5c07e5ccaecca556703c9d0748b8b3ceff0a3393/utils/nlp.py#L146". This modification will increase the BLEU score. If you remove it, I guess you will have a consistent score.

**Comment by tobytyx** [comment]
> Yep, I know that problem. I inherited the evaluation script from https://github.com/budzianowski/multiwoz, their script appends a start token and an end token to the generated sentence "https://github.com/budzianowski/multiwoz/blob/5c07e5ccaecca556703c9d0748b8b3ceff0a3393/utils/nlp.py#L146". This modification will increase the BLEU score. If you remove it, I guess you will have a consistent score.

Thanks a lot!
